src/aioquic/h3/connection.py

```diff
@@ -362,19 +362,19 @@ def send_headers(
             stream_id, encode_frame(FrameType.HEADERS, frame_data), end_stream
         )
 
-    def close_connection(self, last_stream_id: int = None) -> None:
+    def shutdown(self, last_stream_id: int = None) -> None:
         """
-        Close a connection, emitting a GOAWAY frame.
-        :param last_stream_id: Client-initailized stream ID that can be responsed before connection is closed.
+        Send a GOAWAY frame to client from server.
+        :param last_stream_id: Client-initiated stream ID that can be served before connection is closed.
         """
         # client need not send GOAWAY frame
         assert not self._is_client, "Client must not send a goaway frame"
         if last_stream_id is None:
-            last_stream_id = self._max_client_init_bi_stream_id
+            last_stream_id = self._max_client_bidi_stream_id
```

Comment:
What happens if `self._max_client_bidi_stream_id` is None?

Reply:
Maybe we are different concept about client-initiated stream ID.

Client-initiated stream ID means just one of client's request stream IDs. Not push stream ID nor uni-directional stream ID.

H2Connection explain that `last_stream_id` is the last stream which was processed by the sender.

FYI [HTTP3 dfraft](https://tools.ietf.org/html/draft-ietf-quic-http-23#section-5.2)

I'm not very good at English. Any suggestion?

Reply:
Nerver mind about my PRs. I have my own solution for now.